### PR TITLE
[bounty code moment] Adds a way to view the 'Server's Last Round'

### PR DESCRIPTION
**Pull request by MrDoomBringer**
## About The Pull Request
Adds an option to view the last round of a given server, rather than the last round by your own client. This is accomplished by logging the HTML round-end reports, and placing a copy representing the last one in a fixed directory (to be overwritten each shift).

## Why It's Good For The Game
More logging, per headmin request
https://tgstation13.org/phpBB/viewtopic.php?f=41&t=27656#p577600

## Changelog
:cl:
add: You can now view a server's last round (not just yours specifically) via the OOC tab > Server's Last Round
/:cl:

I want to make sure this is done the right way, as modifying logs is a first for me. I've gotten maintainer go-ahead to add the round-end reports to our logs, but I want to make sure this specific implementation for the whole 'track the last round-end report" thing is solid. I fiddled with dynamically finding the last round's report by writing a weird recursive filebrowser algorithm, but so many edge-cases started popping up that I pivoted to storing a copy of the file in a fixed space, which turned out to be much cleaner.

### CODE REVIEW

File: code/__HELPERS/roundend.dm [hunk 1]
```diff
@@ -359,20 +359,48 @@
...
 	var/datum/browser/roundend_report = new(C, "roundend")
 	roundend_report.width = 800
 	roundend_report.height = 600
 	var/content
 	var/filename = C.roundend_report_file()
-	if(!previous)
+	if(your_last_round) //Look at this player's last round
+		content = file2text(filename)
+	else if (server_last_round) //Look at the last round that this server has seen
+		content = file2text("data/last_roundend/server_last_roundend_report.html")
```
Comment (MrDoomBringer) [inline code/__HELPERS/roundend.dm:396]: Let me know if I should have this string stored as a variable somewhere.

File: code/__HELPERS/roundend.dm [hunk 2]
```diff
@@ -359,20 +359,48 @@
...
+	var/content = parts.Join()
+	//Log the rendered HTML in the round log directory
+	fdel(filename)
+	text2file(content, filename)
+	//Place a copy in the root folder, to be overwritten each round. 
+	filename = "data/last_roundend/server_last_roundend_report.html"
+	fdel(filename)
+	text2file(content, filename)
+
+/datum/controller/subsystem/ticker/proc/show_roundend_report(client/C, your_last_round = FALSE, server_last_round = FALSE)
```
Comment (AnturK) [inline code/__HELPERS/roundend.dm:387]: Squish these arguments into single one with defines please. Something like:
```
show_roundend_report(blah, report_type = PERSONAL_LAST_ROUND)
show_roundend_report(blah, report_type = SERVER_LAST_ROUND)
```

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/__HELPERS/roundend.dm (hunk 2 above)
Comment (MrDoomBringer) [reply code/__HELPERS/roundend.dm]: 👍

File: code/__HELPERS/roundend.dm (hunk 1 above)
Comment (AnturK) [reply code/__HELPERS/roundend.dm:398]: What's the point of the directory here.

Comment (MrDoomBringer) [reply code/__HELPERS/roundend.dm]: In my head I thought this would be cleaner for some reason 🥴

**Pull request merged**